src/solver/conv_asm_implicit_gemm_gtc_bwd.cpp

```diff
@@ -163,23 +163,707 @@ GetImplicitGemmGtcDynamicBwdTunablesList()
+        { "bwd",  "fp16",  1,  0,  16,  16,  16,  16,  16,  4,  1,  1,  1,  1,  {1,  2,  2,  1},  {1,  8,  1,  8},  {1,  2,  2,  1},  {1,  8,  1,  8},  0 },  
+        { "bwd",  "fp16",  1,  1,  16,  16,  16,  16,  16,  4,  1,  1,  1,  1,  {2,  1,  1,  2},  {1,  8,  1,  8},  {2,  1,  2,  1},  {1,  8,  1,  8},  0 },  
+        { "bwd",  "fp16",  1,  1,  16,  16,  16,  16,  16,  4,  1,  1,  1,  1,  {2,  1,  2,  1},  {1,  8,  1,  8},  {2,  1,  2,  1},  {1,  8,  1,  8},  0 },  
+        { "bwd",  "fp16",  1,  0,  16,  16,  8,  16,  16,  4,  1,  1,  1,  1,  {1,  1,  1,  2},  {1,  8,  1,  8},  {1,  1,  1,  2},  {1,  8,  1,  8},  0 },  
+        { "bwd",  "fp16",  1,  0,  16,  16,  8,  16,  16,  4,  1,  1,  1,  1,  {1,  1,  2,  1},  {1,  8,  1,  8},  {1,  1,  2,  1},  {1,  8,  1,  8},  0 },  
+        { "bwd",  "fp16",  1,  1,  16,  16,  8,  16,  16,  4,  1,  1,  1,  1,  {1,  1,  1,  2},  {1,  8,  1,  8},  {1,  1,  2,  1},  {1,  8,  1,  8},  0 },  
+        { "bwd",  "fp16",  1,  1,  16,  16,  8,  16,  16,  4,  1,  1,  1,  1,  {1,  1,  2,  1},  {1,  8,  1,  8},  {1,  1,  2,  1},  {1,  8,  1,  8},  0 },  
+        { "bwd",  "fp16",  1,  0,  16,  16,  16,  16,  16,  4,  1,  1,  1,  1,  {1,  1,  1,  4},  {1,  16,  1,  4},  {1,  1,  1,  4},  {1,  16,  1,  4},  0 },  
+        { "bwd",  "fp16",  1,  0,  16,  16,  16,  16,  16,  4,  1,  1,  1,  1,  {1,  1,  4,  1},  {1,  16,  1,  4},  {1,  1,  4,  1},  {1,  16,  1,  4},  0 },  
+        { "bwd",  "fp16",  1,  1,  16,  16,  16,  16,  16,  4,  1,  1,  1,  1,  {1,  1,  1,  4},  {1,  16,  1,  4},  {1,  1,  4,  1},  {1,  16,  1,  4},  0 },  
+        { "bwd",  "fp16",  1,  1,  16,  16,  16,  16,  16,  4,  1,  1,  1,  1,  {1,  1,  4,  1},  {1,  16,  1,  4},  {1,  1,  4,  1},  {1,  16,  1,  4},  0 },  
+        { "bwd",  "fp16",  1,  0,  4,  64,  16,  4,  64,  4,  1,  1,  1,  1,  {1,  1,  1,  1},  {1,  16,  1,  4},  {1,  1,  16,  1},  {1,  16,  1,  4},  0 },  
+        { "bwd",  "fp16",  1,  1,  4,  64,  16,  4,  64,  4,  1,  1,  1,  1,  {1,  1,  1,  1},  {1,  16,  1,  4},  {1,  1,  16,  1},  {1,  16,  1,  4},  0 },  
     };
+
     // clang-format on
-    return tunables;
+    if(ctx.IsFp16())
+        return tunables_fp16;
+    else
+        return tunables_fp32;
 }
 
-static std::tuple<bool,        // is suitable kernel found
+// This is a helper function for selecting better performing config
+static bool mayHaveBiggerGemmNPerBlock(int gemm_m,
+                                       int gemm_n,
+                                       bool is_unit_yx,
+                                       const TunableImplicitGemmGTCDynamic_t& cfg)
+{
+    float n_times_m = static_cast<float>(gemm_n) / static_cast<float>(gemm_m);
+
+    // since gemm_n is much bigger than gemm_m, we want to use a config where gemm_n_per_block is
+    // bigger than gemm_m_per_block
+    if(!is_unit_yx && n_times_m >= 16.0f && cfg.gemm_m_per_block >= cfg.gemm_n_per_block &&
```

Comment:
How to get the 16.f here?

Reply:
Experimental. No theoretical method to fix a threshold for this.